api/src/auth/drivers/openid.ts

```diff
@@ -70,6 +71,12 @@ export class OpenIDAuthDriver extends LocalAuthDriver {
 		this.usersService = new UsersService({ knex: this.knex, schema: this.schema });
 		this.config = additionalConfig;
 
+		const roleMapping: String = env[`AUTH_${config['provider'].toUpperCase()}_ROLE_MAPPING`] as string;
+		if (roleMapping && roleMapping.trim().length > 0) {
+			const roleMap = roleMapping.split(";").map(roleMapItem => roleMapItem.split("=") as [string, string]);
```

Comment:
Is the `key=value;` format for environment variables common / a standard? We usually opt for relying on JSON values for more complicated env variable values, but am happy to go with this if it's the more 'expected' format 🙂

Reply:
Thank you for the review!

I don't think it's a standard, I just thought is was a convenient way (from a usability point of view) for a user to specify a role mapping.
However, if you'd prefer JSON I'm more than happy to change it!

Reply:
Copy that! Let's go with JSON instead. That also means the format will work more 'natively' for the folks who use config files instead of env vars, f.e. in yaml it would be:

```yaml
auth_github_role_mapping:
  admin: 4edb9534-7b9b-47a0-b02d-8876c2ba6311
  maintainer: 4cb2a57c-211f-431b-bfe6-e6411f5eb081
```

Lets go with `{ [provider_key]: [directus_role] }`

Reply:
Just implemented it in commit #[5634b33](https://github.com/directus/directus/pull/24113/commits/5634b33d5bf995bb049f030df0e0ab43d3ce6f4b)